### DISCUSSION TITLE
Unraid template

### DISCUSSION THREAD

**Discussion by Brramble** [henrygd/beszel#52] [General]
Hey, please could this be added to the unraid community store? Or, with your permission could I request it to be added by a maintainer?

**Comment by JamsRepos** [comment]
Just added one to the store, should be available in a couple hours.

**Comment by henrygd** [comment]
I don't use Unraid so I'm 100% fine with it being managed by others as long as there's no security concerns associated with that. And it's able to be updated in the future.

**Comment by JamsRepos** [comment]
Yeah absolutely, it'll auto-update to any latest release. Any new/changes to environment/labels then i'll of course need to update but open to PR's if I miss it.

**Comment by Heavensong89** [comment]
the /var/run/docker.sock should be set to Read-Only ideally, just seen the template drop in the store, saved me having to do my own container setup! 😄

**Comment by JamsRepos** [comment]
> the /var/run/docker.sock should be set to Read-Only ideally, just seen the template drop in the store, saved me having to do my own container setup! 😄

Good catch, i'll update it now. It'll roll out within 2 hours.

**Comment by doomedramen** [comment]
Cheers @jamsrepos

**Comment by henrygd** [comment]
Sounds like this has been worked out. I'm going to close the issue and convert it to a discussion in case anyone wants to talk about it in the future.

If you need anything more from me, let me know.

**Comment by henrygd** [comment]
Another Unraid user opened an issue (#143) where Beszel shows 0% disk usage. Their `df` output also shows 0% disk usage for `/mnt/disk1`, possibly because the filesystem only consists of other mountpoints.

If anyone else ran into this and knows a solution, it would be great if you could point them in the right direction. Otherwise no worries, it's probably something specific to their setup.

**Reply by JamsRepos** [reply]
```
root@Home:~# df
Filesystem       1K-blocks        Used   Available Use% Mounted on
rootfs            32865980      765256    32100724   3% /
tmpfs               131072        7140      123932   6% /run
/dev/sda1         30046848     1147744    28899104   4% /boot
overlay           32865980      765256    32100724   3% /lib
overlay           32865980      765256    32100724   3% /usr
devtmpfs              8192           0        8192   0% /dev
tmpfs             32883072           0    32883072   0% /dev/shm
tmpfs               131072         896      130176   1% /var/log
tmpfs             19729844           0    19729844   0% /var/lib/memtester
tmpfs                 1024           0        1024   0% /mnt/disks
tmpfs                 1024           0        1024   0% /mnt/remotes
tmpfs                 1024           0        1024   0% /mnt/addons
tmpfs                 1024           0        1024   0% /mnt/rootshare
disk1          15485357440  9002989952  6482367488  59% /mnt/disk1
disk2          15485356800  7665504256  7819852544  50% /mnt/disk2
disk4           5397736960         256  5397736704   1% /mnt/disk4
/dev/nvme0n1p1  1953513560   581784908  1369935316  30% /mnt/cache
scratch          942915072   522590976   420324096  56% /mnt/scratch
disk4/data     15485344384 10087607680  5397736704  66% /mnt/disk4/data
shfs           46456110720 26756153984 19699956736  58% /mnt/user0
shfs           46456110720 26756153984 19699956736  58% /mnt/user
/dev/loop2       104857600    19368532    84636780  19% /var/lib/docker
/dev/loop3         1048576        4780      925556   1% /etc/libvirt
```

![image](https://github.com/user-attachments/assets/496fc70c-f04c-4a6a-9791-45be9133de22)

Hope this helps.

**Reply by death4444** [reply]
> ```
> root@Home:~# df
> Filesystem       1K-blocks        Used   Available Use% Mounted on
> rootfs            32865980      765256    32100724   3% /
> tmpfs               131072        7140      123932   6% /run
> /dev/sda1         30046848     1147744    28899104   4% /boot
> overlay           32865980      765256    32100724   3% /lib
> overlay           32865980      765256    32100724   3% /usr
> devtmpfs              8192           0        8192   0% /dev
> tmpfs             32883072           0    32883072   0% /dev/shm
> tmpfs               131072         896      130176   1% /var/log
> tmpfs             19729844           0    19729844   0% /var/lib/memtester
> tmpfs                 1024           0        1024   0% /mnt/disks
> tmpfs                 1024           0        1024   0% /mnt/remotes
> tmpfs                 1024           0        1024   0% /mnt/addons
> tmpfs                 1024           0        1024   0% /mnt/rootshare
> disk1          15485357440  9002989952  6482367488  59% /mnt/disk1
> disk2          15485356800  7665504256  7819852544  50% /mnt/disk2
> disk4           5397736960         256  5397736704   1% /mnt/disk4
> /dev/nvme0n1p1  1953513560   581784908  1369935316  30% /mnt/cache
> scratch          942915072   522590976   420324096  56% /mnt/scratch
> disk4/data     15485344384 10087607680  5397736704  66% /mnt/disk4/data
> shfs           46456110720 26756153984 19699956736  58% /mnt/user0
> shfs           46456110720 26756153984 19699956736  58% /mnt/user
> /dev/loop2       104857600    19368532    84636780  19% /var/lib/docker
> /dev/loop3         1048576        4780      925556   1% /etc/libvirt
> ```
> 
> ![图像](https://private-user-images.githubusercontent.com/1347620/363279859-496fc70c-f04c-4a6a-9791-45be9133de22.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwOTM0MzgsIm5iZiI6MTcyNTA5MzEzOCwicGF0aCI6Ii8xMzQ3NjIwLzM2MzI3OTg1OS00OTZmYzcwYy1mMDRjLTRhNmEtOTc5MS00NWJlOTEzM2RlMjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMDgzMjE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Mzk0MjUyNjA2MzgwMDg0NTk1MTY0MmRiMDUxOTc4NmNlYmUyMGUyMjZiNjA4YzkyMTM2NzZiZWUwMjAwYWE1MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bYUc_58MgOoBoulv6mOQkT7E9q9yVkXm4MtoPZ117qg)
> 
> 希望这有帮助。

WOW，How?
How is the FILESYSTEM parameter filled in?

**Reply by JamsRepos** [reply]
![image](https://github.com/user-attachments/assets/6abb80cd-f257-4630-bb47-e9d9a633cb47)

its set to my cache.

**Comment by jasontucker** [comment]
I'm wondering if it would be better for this to run as a plugin in Unraid than a docker container for the agent since agents in docker cant see the GPU this may be a better work around with using the binary as a plugin.

**Reply by D3r3k23** [reply]
I agree that it would be useful, since GPU monitoring isn't available in the agent docker container. I did a bit of searching but didn't find much information on how to create a plugin. @JamsRepos is this something you could do?

**Comment by ransom88** [comment]
On my Unraid machine (Lincstation N2 with three m.2 nvme 4TB ssds in RAIDZ1) I have installed the docker Beszel agent. No matter what I set FILESYSTEM to it only ever picks up "loop3" which is a 40GB device about 25% full (~10GB). What I would ideally want is to know how % full my main storage (/mnt/storage) which is about 19% full (1.28 TB out of 6.76 TB).

If I set LOG_LEVEL=DEBUG I get the following lines
```
DEBUG Disk partitions=
"[
{\"device\":\"/dev/loop3\",\"mountpoint\":\"/\",\"fstype\":\"btrfs\",\"opts\":[\"rw\",\"noatime\",\"bind\"]} 
{\"device\":\"/dev/loop3\",\"mountpoint\":\"/etc/resolv.conf\",\"fstype\":\"btrfs\",\"opts\":[\"rw\",\"noatime\",\"bind\"]} 
{\"device\":\"/dev/loop3\",\"mountpoint\":\"/etc/hostname\",\"fstype\":\"btrfs\",\"opts\":[\"rw\",\"noatime\",\"bind\"]} 
{\"device\":\"/dev/loop3\",\"mountpoint\":\"/etc/hosts\",\"fstype\":\"btrfs\",\"opts\":[\"rw\",\"noatime\",\"bind\"]}
]"
```

And some variation of this:

```
2025/08/18 15:32:41 WARN Partition details not found filesystem=/dev/nvme1n1p1
2025/08/18 15:32:41 INFO Detected root device name=loop3
```

For reference,

```
df -h
Filesystem                Size  Used Avail Use% Mounted on
rootfs                    7.7G  332M  7.4G   5% /
tmpfs                     128M  2.0M  127M   2% /run
/dev/sda1                 3.8G  1.7G  2.2G  44% /boot
overlay                   7.7G  332M  7.4G   5% /usr
overlay                   7.7G  332M  7.4G   5% /lib
tmpfs                     128M  336K  128M   1% /var/log
devtmpfs                  8.0M     0  8.0M   0% /dev
tmpfs                     7.8G     0  7.8G   0% /dev/shm
efivarfs                  192K   86K  102K  46% /sys/firmware/efi/efivars
tmpfs                     1.0M     0  1.0M   0% /mnt/disks
tmpfs                     1.0M     0  1.0M   0% /mnt/remotes
tmpfs                     1.0M     0  1.0M   0% /mnt/addons
tmpfs                     1.0M     0  1.0M   0% /mnt/rootshare
storage                   6.2T  256K  6.2T   1% /mnt/storage
storage/isos              6.2T  128K  6.2T   1% /mnt/storage/isos
storage/domains           6.2T  128K  6.2T   1% /mnt/storage/domains
storage/doylehaourphotos  6.5T  319G  6.2T   5% /mnt/storage/doylehaourphotos
storage/system            6.2T   11G  6.2T   1% /mnt/storage/system
storage/Mike Backup       6.2T   39G  6.2T   1% /mnt/storage/Mike Backup
storage/pp                6.2T  652M  6.2T   1% /mnt/storage/pp
storage/appdata           6.2T  2.2G  6.2T   1% /mnt/storage/appdata
storage/rjd               6.4T  170G  6.2T   3% /mnt/storage/rjd
storage/films             6.8T  654G  6.2T  10% /mnt/storage/films
/dev/loop2                1.0G  6.1M  905M   1% /etc/libvirt
/dev/loop3                 40G  9.7G   30G  25% /var/lib/docker
storage/.beszel           6.2T  128K  6.2T   1% /mnt/storage/.beszel


lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0 579.8M  1 loop /usr
loop1         7:1    0 169.8M  1 loop /lib
loop2         7:2    0     1G  0 loop /etc/libvirt
loop3         7:3    0    40G  0 loop /var/lib/docker/btrfs
                                      /var/lib/docker
sda           8:0    1   3.8G  0 disk 
└─sda1        8:1    1   3.7G  0 part /boot
nvme2n1     259:0    0   3.7T  0 disk 
└─nvme2n1p1 259:4    0   3.7T  0 part 
nvme1n1     259:1    0   3.7T  0 disk 
└─nvme1n1p1 259:5    0   3.7T  0 part 
nvme0n1     259:2    0   3.7T  0 disk 
└─nvme0n1p1 259:3    0   3.7T  0 part

```

Anyone have any ideas for anything I can try?

**Reply by henrygd** [reply]
Your docker storage is on loop3 so that's all it can see. You would need to give the container access to another partition by mounting it in the container.

I'm not sure how unraid works or what you have access to, since `lsblk` doesn't show a mountpoint for `nvme1n1p1`. You could try something like this:

```yaml
beszel-agent:
  volumes:
    - /mnt/storage/.beszel:/extra-filesystems/nvme1n1p1:ro
```

What you're looking for is another partition to show up in the `DEBUG Disk partitions` log.

**Comment by ransom88** [comment]
Thank you so much for your helpful response. I'll have a look at this when I have time and hopefully I now have what I need to make some progress.
